### ISSUE TITLE
propagationPolicy for a resource

### ISSUE THREAD

**Issue by npnwdev** [kubernetes/kubernetes#107147]
### What happened?

my controller creates two resources `AA` and `BB` where `AA` is the parent and `BB` is the child.
Child `BB` is configured with `ownerReferences:` pointing to `AA` and with `blockOwnerDeletion: true`

If I understand correctly, propagationPolicy is `background` by default.  
So when `AA` is deleted, `AA` is removed immediately and `BB` is left to be cleaned by GC.

However, in my case, the controller is not expecting the deletion of `AA`. 
When it sees `AA` is deleted, it immediately tries to recreate `AA`. Now `BB` is not cleaned by GC by that time and is never deleted. I guess it's not deleting `BB` because `AA` is recreated.

After this, I've got `new AA` + `old BB` while the expectation is `new AA` + `new BB`. 
This creates an inconsistency.

As far as I checked, I couldn't find a way to define `propagationPolicy` when defining a resource except during `kubectl delete`.

why did GC never delete `BB`? 
IMO `new AA` shouldn't be considered as the owner of `BB`. let me know, please. Thanks!

### What did you expect to happen?

Child resources should be deleted when their parent is deleted. In this case, `BB` is never deleted.

### How can we reproduce it (as minimally and precisely as possible)?

1. Create two resources, a parent and a child.
2. create a watch for a parent, recreate parent resource immediately whenever a parent is deleted
3. delete the parent

### Kubernetes version

<details>
```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.5", GitCommit:"6b1d87acf3c8253c123756b9e61dac642678305f", GitTreeState:"clean", BuildDate:"2021-03-18T01:10:43Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.0", GitCommit:"af46c47ce925f4c4ad5cc8d1fca46c7b77d13b38", GitTreeState:"clean", BuildDate:"2020-12-08T17:51:19Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```
</details>

### Cloud provider

<details>
Not Applicable
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

$ uname -a
Linux myvm 3.10.0-1160.el7.x86_64 #1 SMP Mon Oct 19 16:18:59 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```
</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by caesarxuchao** [comment]
The GC internally tracks objects based on their UIDs, not their names, so recreating AA shouldn't affect the GC of BB.

Does BB have other owners?

**Comment by npnwdev** [comment]
@caesarxuchao Sorry about the delay. 
BB doesn't have any other owners.

**Comment by npnwdev** [comment]
found a strange problem with object `BB`. I see that the object BB was never deleted but it has ownerReference to new-AA. I guess something wrong with my controller. Will update my findings shortly

**Comment by npnwdev** [comment]
I guess it's the propagation delay. before object BB is deleted, my controller tries to create one (assuming it'd have been removed by GC) so it ends up updating object BB with ownerReference to new-AA and BB never gets deleted.

I tried deleting with `--cascade=foreground` and found `BB` is always deleted.
So I guess the problem can be fixed by using the `foreground` policy.

I'm not sure how i can set up the propagationPolicy for the customResource (`BB`) i.e instead of specifying during delete command
